client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/configuration/base/ResyncConfigurationManager.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2014-2015 CyberVision, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kaaproject.kaa.client.configuration.base;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import javax.annotation.Generated;
+
+import org.kaaproject.kaa.client.KaaClientProperties;
+import org.kaaproject.kaa.schema.base.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Generated("ResyncConfigurationManager.java.template")
+public class ResyncConfigurationManager extends AbstractConfigurationManager implements ConfigurationManager{
+    
+    private static final Logger LOG = LoggerFactory.getLogger(AbstractConfigurationManager.class);
+    
+    public ResyncConfigurationManager(KaaClientProperties properties) {
+        super(properties);
+    }
+
+    @Override
+    public Configuration getConfiguration() {
+        try {
+            return deserializer.fromByteArray(getConfigurationData());
```

Comment:
Looks like the deserialization will be processed each time I call the getConfiguration(). Maybe it would be better to cache the deserialization result?